tests/i18n/test_extraction.py

```diff
@@ -127,6 +127,33 @@ def assertNotRecentlyModified(self, path):
         self.assertGreater(delta, 10, "%s wasn't recently modified" % path)
 
 
+class NormalizeLocaleTest(ExtractorTests):
+
+    LOCALES = ('zh-cn', 'zh_cn', 'zh_CN', 'ZH-CN')
+    LOCALE = 'zh_CN'
+    PO_FILE = 'locale/%s/LC_MESSAGES/django.po' % LOCALE
+
+    def test_normalize_locale(self):
+        """
+        normalize the different strings used for locales.
+        le (zh-cn, zh_cn, zh_CN, ZH-CN) all these strings will be to zh_CN
+        """
+        out = StringIO()
+        for locale in self.LOCALES:
+            locale_normalized = normalize_locale(locale, out)
+            self.assertEqual(self.LOCALE, locale_normalized)
+
+    def test_makemessages_with_normalized_locale(self):
+        """
+        Test the makemessages works with normalized locales in the input.
+        """
+        out = StringIO()
+        for locale in self.LOCALES:
+            locale_normalized = normalize_locale(locale, out)
+            management.call_command('makemessages', locale=[locale_normalized], verbosity=0)
```

Comment:
I thought the whole point was that `makemessages` normalized the locale within. This test is incorrectly doing the normalization up front.

Reply:
Ah. Thanks for the catch.